src/hypothesis/internal/conjecture/engine.py

```diff
@@ -267,6 +269,44 @@ def test_function(self, data):
 
         self.record_for_health_check(data)
 
+    def generate_novel_prefix(self):
+        prefix = bytearray()
+        node = 0
+        while True:
+            assert len(prefix) < self.cap
+            assert node not in self.dead
+            upper_bound = self.capped.get(node, 255) + 1
+            try:
+                c = self.forced[node]
+                prefix.append(c)
+                node = self.tree[node][c]
+                continue
+            except KeyError:
+                pass
+            c = self.random.randrange(0, upper_bound)
+            try:
+                next_node = self.tree[node][c]
+                if next_node in self.dead:
+                    choices = [
+                        c for c in hrange(upper_bound)
```

Comment:
I'd use a different comprehension variable here (`i`?), due to the variation in scoping rules in various Pythons.  I don't think it makes a difference here even with a reference in the `except` block, but easy to avoid the need to check 😄